component/module/string/string.go

```diff
@@ -0,0 +1,160 @@
+// Package string defines the module.string component.
+package string
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/flow"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+	"github.com/grafana/agent/pkg/flow/tracing"
+	"github.com/grafana/agent/web/api"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "module.string",
+		Args:    Arguments{},
+		Exports: Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments holds values which are used to configure the module.string
+// component.
+type Arguments struct {
+	// Content to load for the module.
+	Content rivertypes.Secret `river:"content,attr"`
+
+	// Arguments to pass into the module.
+	Arguments map[string]any `river:"arguments,attr,optional"`
+}
+
+// Exports holds values which are exported from the run module.
+type Exports struct {
+	// Exports exported from the running module.
+	Exports map[string]any `river:"exports,attr"`
+}
```

Comment:
@erikbaranowski I would like to make both arguments and exports blocks before we release, but given #3244 isn't fixed yet, should we merge these as attributes and revisit in a follow up PR once #3244 is fixed?

Reply:
I'd merge as attributes for now. Since we know there is more modules work to do we risk scope creeping this PR